### DISCUSSION TITLE
(Rust) For those who work on exercises in an external IDE, how do you organize your solutions?

### DISCUSSION THREAD

**Discussion by zGrvs** [codewars/codewars.com#3185] [Q&A]
I'm new to Rust so apologies for the noob question. I have a CodeWars directory in which I keep subdirectories for each language I'm practicing. Organizing C was easy; in my C directory I keep further sub directories for each exercise where I code my main.c file and store a build if I end up debugging. Autocomplete worked as expected, everything was fine.

Rust is proving more difficult. From what I understand, some of the files created when you run "cargo new"- such as the toml file- are required for autocomplete to work correctly. If I start working out a solution in the default main.rs file created, things are good. However, the moment I try creating a new .rs file for another exercise, autocomplete no longer works. My plan was to have just one cargo project where I would create different .rs files in the src directory for each exercise, but clearly this is not working. I'm also wondering how that would affect my ability to debug if every exercise I work on is inside the same cargo project.

So anyone who works on Rust exercises externally; how do you do it? Should I create an entire cargo project for each exercise? It seems this would be somewhat redundant and take up significantly more storage space. Maybe I should ditch the desire to keep all of my solutions archived on my hard drive?

**Comment by Invariance-NaN** [comment]
Try adding the files as modules to your `main.rs` (with `mod file_name;`). That should get the analysis tools working (or at least it does in VSCode).

**Reply by zGrvs** [reply]
I hadn’t thought about that, thank you!

**Reply by Invariance-NaN** [reply]
Happy to help!
